File: Editorial/1101-1200/1121_Divide_Array_Into_Increasing_Sequences/Go/1_Approach_Maximum_Value.go
func canDivideIntoSubsequences(nums []int, k int) bool {
	if k == 1 {
		return true
	}
	pre := nums[0]
	cnt := 0
	for i := 0; i < len(nums); i++ {
		if pre == nums[i] {
			cnt++
		} else {
			if cnt*k > len(nums) {
				return false
			}
			pre = nums[i]
			cnt = 1
		}
	}
	return cnt*k <= len(nums)
}

File: Editorial/1101-1200/1121_Divide_Array_Into_Increasing_Sequences/JavaScript/1_Approach_Maximum_Value.js
var canDivideIntoSubsequences = function (nums, k) {
    let pre = nums[0],
        cnt = 0;
    for (let n of nums) {
        if (n === pre) {
            cnt++;
        } else {
            pre = n;
            cnt = 1;
        }
        if (cnt * k > nums.length) {
            return false;
        }
    }
    return true;
};
